server/functional-test/src/test/java/org/zanata/feature/testharness/ZanataTestCase.java

```diff
@@ -59,6 +62,18 @@ protected void before() throws Throwable {
 
         @Override
         protected void after() {
+            // Close all windows other than the original one
+            WebDriver driverRef = WebDriverFactory.INSTANCE.getDriver();
+            List<String> windows = Lists.newArrayList(driverRef.getWindowHandles());
+            String mainWindowHandle = windows.get(0);
+            for (String windowHandle : windows) {
+                if (!windowHandle.equals(mainWindowHandle)) {
```

Comment:
Couldn't you just loop i from 1 (not the usual 0) while i < windows.size()?

    for (int i = 1; i < windows.size(), i++) {